### PR TITLE
feat: add spring config metadata

### PR DESCRIPTION
**Pull request by loong024**
add spring config metadata

### CODE REVIEW

File: spring-cloud-alibaba-examples/spring-cloud-ai-example/README-en.md [hunk 1]
```diff
@@ -33,7 +33,7 @@ The Spring Cloud Alibaba AI module is based on [Spring AI 0.8.1](https://docs.sp
    spring:
      cloud:
        ai:
-         tongyi:
+         tong-yi:
```
Comment (yuluo-yx) [inline spring-cloud-alibaba-examples/spring-cloud-ai-example/README-en.md]: why was this changed and have you tested this locally?

File: spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-ai/pom.xml [hunk 2]
```diff
@@ -61,6 +61,12 @@
 			<optional>true</optional>
 		</dependency>
 
+		<dependency>
```
Comment (yuluo-yx) [inline spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-ai/pom.xml:64]: Can you explain why you are adding this dependency and what are the implications if you don't? 🤔

Comment (loong024) [reply spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-ai/pom.xml:64]: 一. Why add this dependency
1. During compilation, a metadata file describing configuration properties will be generated for classes annotated with @ConfigurationProperties;
2. The generated configuration metadata will be recognized by the IDE, thereby providing code completion features
二. What will be the impact if this dependency is not added?
1. IDE cannot provide automatic completion and type safety checking for the customized @ConfigurationProperties class. I need to find the name and type of the corresponding configuration property myself.
2. The configuration metadata file (META-INF/spring-configuration-metadata.json) will not be generated in the jar package.

File: spring-cloud-alibaba-examples/spring-cloud-ai-example/README-en.md (hunk 1 above)
Comment (loong024) [reply spring-cloud-alibaba-examples/spring-cloud-ai-example/README-en.md]: Typo prompted by ide: In word 'tongyi' It seems that it is not a big problem, I withdraw it

**Review by yuluo-yx — APPROVED** [review]
LGTM, thanks
